src/status_im/chat/events/commands.cljs

```diff
@@ -0,0 +1,122 @@
+(ns status-im.chat.events.commands
+  (:require [cljs.reader :as reader]
+            [clojure.string :as str]
+            [re-frame.core :refer [reg-fx reg-cofx inject-cofx dispatch trim-v]]
+            [taoensso.timbre :as log]
+            [status-im.data-store.messages :as msg-store]
+            [status-im.utils.handlers :refer [register-handler-fx]]
+            [status-im.components.status :as status]
+            [status-im.chat.constants :as const]
+            [status-im.commands.utils :as commands-utils]
+            [status-im.i18n :as i18n]
+            [status-im.utils.platform :as platform]))
+
+;;;; Helper fns
+
+(defn generate-context
+  "Generates context for jail call"
+  [{:keys [current-account-id chats]} chat-id to]
+  (merge i18n/delimeters
+         {:platform platform/platform
+          :from     current-account-id
+          :to       to
+          :chat     {:chat-id chat-id
+                     :group-chat (get-in chats [chat-id :group-id])}}))
+
+;;;; Coeffects
+
+(reg-cofx
+ ::get-persisted-message
+ (fn [coeffects _]
+   (assoc coeffects :get-persisted-message msg-store/get-by-id)))
+
+;;;; Effects
+
+(reg-fx
+ ::update-persisted-message
+ (fn [message]
+   (msg-store/update message)))
+
+
+(reg-fx
+ :chat-fx/call-jail
+ (fn [{:keys [callback-events-creator] :as opts}]
+   (status/call-jail
+    (-> opts
+        (dissoc :callback-events-creator)
+        (assoc :callback
+               (fn [jail-response]
+                 (doseq [event (callback-events-creator jail-response)]
+                   (dispatch event))))))))
+
+;;;; Handlers
+
+(register-handler-fx
+ ::jail-command-data-response
+ [trim-v]
```

Comment:
It might be worth discussing about this trim-v middleware, personally I am against it because unless it is used for all handlers it makes an unnecessary second syntax for the same thing (I'm a big fan of homogeneity in big codebase with many contributors)